CRM/Contribute/BAO/ContributionRecur.php

```diff
@@ -94,6 +96,43 @@ public static function add(&$params) {
     return $recurring;
   }
 
+  /**
+   * Event fired after modifying a contribution.
+   * @param \Civi\Core\Event\PostEvent $event
+   */
+  public static function self_hook_civicrm_post(\Civi\Core\Event\PostEvent $event) {
+    if ($event->action === 'edit') {
+      if (is_numeric($event->object->amount)) {
+        $templateContribution = CRM_Contribute_BAO_ContributionRecur::getTemplateContribution($event->object->id);
+        if (empty($templateContribution['id']) || isset(\Civi::$statics[__CLASS__]['update'][$templateContribution['id']])) {
+          // Precaution against a loop - although in theory the lookups provide this
+          // it still seems worth an extra check.
+          return;
+        }
+        $lines = LineItem::get(FALSE)
+          ->addWhere('contribution_id', '=', $templateContribution['id'])
+          ->addWhere('contribution_id.is_template', '=', TRUE)
+          ->addSelect('contribution_id.total_amount')
+          ->execute();
+        if (count($lines) === 1) {
+          $contributionAmount = $lines->first()['contribution_id.total_amount'];
+          if (Money::of($contributionAmount, 'USD')->compareTo($event->object->amount)) {
+            \Civi::$statics[__CLASS__]['update'][$templateContribution['id']] = $templateContribution['id'];
+            // If different then we need to update
+            // the contribution. Note that if this is being called
+            // as a result of the contribution having been updated then there will
+            // be no difference.
+            Contribution::update(FALSE)
+              ->addWhere('id', '=', $templateContribution['id'])
+              ->setValues(['total_amount' => $event->object->amount])
+              ->execute();
+            unset(\Civi::$statics[__CLASS__]['update'][$templateContribution['id']]);
```

Comment:
Noting the static here, I've found that statics are not enough to prevent loops if you have two code paths executing near simultaneously. In this case it's highly likely to be ineffective because the static is set after making API calls. If we need a lock around this it needs to use the `Civi::lockManager()->acquire()` mechanism

Reply:
I was worried about a loop not a deadlock - the deadlock risk is the same as any other bit of code

Reply:
I'm only worried about a php loop - it should stop that -  as any static var will - I don't even know if it's possible to hit

Reply:
OK - I'll pull out the whole static thing - it was a precaution only against a theoretical situation whereby the hook winds up in a code loop - but I never had a practical scenario